module/validation/seal_validator.go

```diff
@@ -0,0 +1,286 @@
+package validation
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+// DefaultRequiredChunkApprovals is the default number of approvals that should be
+// present and valid for each chunk. Setting this to 0 will disable counting of chunk approvals
+// this can be used temporarily to ease the migration to new chunk based sealing.
+// TODO:
+//   * This value is for the happy path (requires just one approval per chunk).
+//   * Full protocol should be +2/3 of all currently staked verifiers.
+const DefaultRequiredChunkApprovals = 1
+
+// sealValidator holds all needed context for checking seal
+// validity against current protocol state.
+type sealValidator struct {
+	state                  protocol.State
+	assigner               module.ChunkAssigner
+	verifier               module.Verifier
+	seals                  storage.Seals
+	headers                storage.Headers
+	payloads               storage.Payloads
+	requiredChunkApprovals uint
+}
+
+func NewSealValidator(state protocol.State, headers storage.Headers, payloads storage.Payloads, seals storage.Seals,
+	assigner module.ChunkAssigner, verifier module.Verifier, requiredChunkApprovals uint) *sealValidator {
+	rv := &sealValidator{
+		state:                  state,
+		assigner:               assigner,
+		verifier:               verifier,
+		headers:                headers,
+		seals:                  seals,
+		payloads:               payloads,
+		requiredChunkApprovals: requiredChunkApprovals,
+	}
+
+	return rv
+}
+
+func (s *sealValidator) verifySealSignature(aggregatedSignatures *flow.AggregatedSignature,
+	chunk *flow.Chunk, executionResultID flow.Identifier) error {
+	// TODO: replace implementation once proper aggregation is used for Verifiers' attestation signatures.
+	for i, signature := range aggregatedSignatures.VerifierSignatures {
+		signerId := aggregatedSignatures.SignerIDs[i]
+
+		nodeIdentity, err := identityForNode(s.state, chunk.BlockID, signerId)
+
+		if err != nil {
+			return err
+		}
+
+		atst := flow.Attestation{
+			BlockID:           chunk.BlockID,
+			ExecutionResultID: executionResultID,
+			ChunkIndex:        chunk.Index,
+		}
+		atstID := atst.ID()
```

Comment:
I think you could move this code block _before_ the start of the for loop https://github.com/onflow/flow-go/blob/033e0be0b881b2b2d19687e97465c524a1fdb21e/module/validation/seal_validator.go#L53